src/Tie/Name.hs

```diff
@@ -313,3 +314,9 @@ extractHaskellModule =
           xs ->
             [Text.intercalate "." xs]
    in concatMap extractModule . Text.words
+
+camelCaseIfNotAllCapitalized :: String -> String
+camelCaseIfNotAllCapitalized word =
+  if all (==True) $ isUpper <$> word
+  then word
+  else toCamelCase word
```

Comment:
Can you check if this has any effect at all? I can't quite pinpoint why the NISEValue would get into NiseValue earlier.

Reply:
You are right, sir. It was `camel` from `casing` package that I was using previously that caused that. `toCamelCase` is fine to use